test/bidder_submitter.test.ts

```diff
@@ -0,0 +1,276 @@
+          name: 'test-filler',
+          keypair: Keypair.random(),
+          minProfitPct: 0,
+          minHealthFactor: 0,
+          forceFill: false,
+          supportedBid: [],
+          supportedLot: [],
+        },
+        auctionEntry: {
+          user_id: 'test-user',
+          auction_type: AuctionType.Liquidation,
+          filler: 'test-filler',
+          start_block: 900,
+          fill_block: 1000,
+        } as AuctionEntry,
+      };
+
+      const result = await bidderSubmitter.submit(submission);
+
+      expect(result).toBe(true);
+      expect(mockLoadAuction).toHaveBeenCalledWith('test-user', AuctionType.Liquidation);
+      expect(mockSubmitTransaction).toHaveBeenCalled();
+      expect(mockDb.setFilledAuctionEntry).toHaveBeenCalled();
+    });
+
+    it('should handle auction already filled', async () => {
+      const mockLoadAuction = jest.fn().mockResolvedValue(undefined);
+
+      (SorobanHelper as jest.MockedClass<typeof SorobanHelper>).mockImplementation(
+        () =>
+          ({
+            loadAuction: mockLoadAuction,
+            network: { rpc: 'mock-rpc', opts: {} },
+          }) as unknown as SorobanHelper
+      );
+
+      const submission: AuctionBid = {
+        type: BidderSubmissionType.BID,
+        filler: {
+          name: 'test-filler',
+          keypair: Keypair.random(),
+          minProfitPct: 0,
+          minHealthFactor: 0,
+          forceFill: false,
+          supportedBid: [],
+          supportedLot: [],
+        },
+        auctionEntry: {
+          user_id: 'test-user',
+          auction_type: AuctionType.Liquidation,
+        } as AuctionEntry,
+      };
+
+      const result = await bidderSubmitter.submit(submission);
+
+      expect(result).toBe(true);
+      expect(mockDb.deleteAuctionEntry).toHaveBeenCalledWith('test-user', AuctionType.Liquidation);
+    });
+  });
+
+  describe('containsAuction', () => {
+    it('should return true if auction is in the queue', () => {
+      const auctionEntry: AuctionEntry = {
+        user_id: 'test-user',
+        auction_type: AuctionType.Liquidation,
+      } as AuctionEntry;
+
+      bidderSubmitter.addSubmission(
+        {
+          type: BidderSubmissionType.BID,
+          auctionEntry: auctionEntry,
+        } as AuctionBid,
+        1
+      );
+
+      expect(bidderSubmitter.containsAuction(auctionEntry)).toBe(true);
+    });
+
+    it('should return false if auction is not in the queue', () => {
+      const auctionEntry: AuctionEntry = {
+        user_id: 'test-user',
+        auction_type: AuctionType.Liquidation,
+      } as AuctionEntry;
+
+      bidderSubmitter['submissions'] = [];
+
+      expect(bidderSubmitter.containsAuction(auctionEntry)).toBe(false);
+    });
+  });
+
+  describe('onDrop', () => {
+    let mockedSendSlackNotif = sendSlackNotification as jest.MockedFunction<
+      typeof sendSlackNotification
+    >;
+    const loggerSpy = jest.spyOn(logger, 'error').mockImplementation();
```

Comment:
It is more consistent to just mock the logger manually. see `pool_event_handler.test.ts` for an example